### PR TITLE
Prepare sending sym/asym deposit transactions

### LINKED ISSUE #537: Liquidity: Add business logic to send "Add Liquidity" transactions

### PR DESCRIPTION
**Pull request by veado**
- [x] Prepare `AddStake` component for sending deposit transactions
- [x] Prepare services to handle deposit transactions
- [x] Create deposit memo depending on `StakeType`
- [x] Fix usage of `sym`/ `asym` states in components + other places (we have used it the other way around before)

Part of #537

### CODE REVIEW

File: src/renderer/services/chain/fees/stake.ts [hunk 1]
```diff
@@ -67,28 +94,32 @@ const stakeFeeByChain$ = (chain: Chain): FeeLD => {
...
-          ),
-          RxOp.map(sequenceTRDFromArray),
-          liveData.map(([base, cross]) => ({
-            base,
-            cross: O.fromNullable(cross)
-          }))
-        )
-      ),
-      O.getOrElse((): StakeFeesLD => Rx.of(RD.initial))
+// TODO (@Veado) Store results of deposit txs into a state, so views will have access to it.
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/chain/fees/stake.ts:97]: "Store results of deposit `txs`(why txs ? maybe fees ?) into a state, so views will have access to it."

if yes we can just create new `observableState" for type
```ts
const {get$: type$, set: setType} = createObservableState()
```
and after just use it as
```ts
const stakeFees$: StakeFeesLD = combineLatest(selectedPoolChain$, type$) // ... and all prev logic here
```
and at the `AddStakeView.tsx`
just 
```ts
useEffect(() => setType(type))
```

what do you think ?

File: src/renderer/services/chain/memo.ts [hunk 2]
```diff
@@ -4,45 +4,37 @@ import * as O from 'fp-ts/lib/Option'
...
+import { runeAddress$, assetAddress$ } from './address'
+import { AsymDepositMemoRx, MemoRx } from './types'
 
 /**
- * Stake memo for txs sent on base-chain
+ * Memo of symmetrical deposit txs
  */
-const baseChainStakeMemo$: MemoRx = Rx.combineLatest([selectedPoolAsset$, crossAddress$]).pipe(
-  RxOp.map(([oPoolAsset, oCrossAddress]) =>
+const symDepositTxMemo$: AsymDepositMemoRx = Rx.combineLatest([selectedPoolAsset$, runeAddress$, assetAddress$]).pipe(
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/chain/memo.ts:15]: seems naming here is a little bit confusing:
```ts
const symDepositTxMemo$: AsymDepositMemoRx 
```
and
```ts
const asymDepositTxMemo$: MemoRx
```

File: src/renderer/services/chain/memo.ts [hunk 3]
```diff
@@ -4,45 +4,37 @@ import * as O from 'fp-ts/lib/Option'
...
 
+const symDepositAssetTxMemo$: MemoRx = symDepositTxMemo$.pipe(RxOp.map(FP.flow(O.map(({ asset }) => asset))))
+
 /**
- * Stake memo for txs sent on cross-chain
+ * Memo of asymmetrical deposit txs
  */
-const crossChainStakeMemo$: MemoRx = Rx.combineLatest([selectedPoolAsset$, baseAddress$]).pipe(
-  RxOp.map(([oPoolAsset, oBaseAddress]) =>
+const asymDepositTxMemo$: MemoRx = selectedPoolAsset$.pipe(
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/chain/memo.ts:33]: seems just 
```ts
const asymDepositTxMemo$: MemoRx = selectedPoolAsset$.pipe(RxOp.map(O.map(getDepositMemo)))
```
is enough

File: src/renderer/services/midgard/pools.ts [hunk 4]
```diff
@@ -255,6 +256,16 @@ const createPoolsService = (
     RxOp.retry(MIDGARD_MAX_RETRY)
   )
 
+  const poolAddress$: PoolAddressRx = poolAddresses$.pipe(
+    RxOp.map(
+      FP.flow(
+        RD.toOption,
+        O.chain(A.head),
+        O.chain(({ address }) => O.fromNullable(address))
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/midgard/pools.ts:264]: [nitpick] there is interesting operator at the `Option` for such cases
```ts
O.mapNullable(head => head.address)
```

**Review by thatStrangeGuyThorchain — CHANGES_REQUESTED** [review]

File: src/renderer/components/stake/add/AddStake.tsx [hunk 5]
```diff
@@ -36,37 +36,44 @@ type Props = {
...
+    poolAddress: oPoolAddress,
+    assets,
+    priceAsset,
+    reloadFees,
+    fees,
+    onChangeAsset,
+    disabled = false,
+    poolData
+  } = props
+
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/components/stake/add/AddStake.tsx:76]: Haha, don't ask `why` (or should I call it "force of habit" or something like that :sweat_smile: )

File: src/renderer/services/bitcoin/fees.ts [hunk 6]
```diff
@@ -47,31 +46,56 @@ export const createFeesService = (oClient$: Client$): FeesService => {
...
+  /**
+   * Factory to create a stream of fees for pool transacions
+   * @param memo Memo used for pool transactions
+   */
+  const poolFeeRate$ = (memo: Memo): FeeRateLD =>
+    FP.pipe(
+      reloadStakeFee$,
+      switchMap(() => memoFees$(memo)),
+      liveData.map((fees) => fees.fast.feeRate),
+      // we do need to store result in a subject to access it w/o subscribing a stream
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/services/bitcoin/fees.ts:90]: In [`sendStakeTx` in `renderer/services/chain/transaction.ts`](https://github.com/thorchain/asgardex-electron/blob/eac53245b9c812b2395586fba255267de8bbea6f/src/renderer/services/chain/transaction.ts#L31) we have to access to current fee rate, but w/o a subscribtion to the stream.

File: src/renderer/services/chain/fees/stake.ts (hunk 1 above)
Comment (veado) [reply src/renderer/services/chain/fees/stake.ts:97]: Yep, something like this. We will implement it later while finishing all needed stuff for deposit txs #537

File: src/renderer/services/binance/transaction.ts [hunk 7]
```diff
@@ -51,9 +51,12 @@ const tx$ = ({
     startWith(RD.pending)
   )
 
-const pushTx = (clientState$: BinanceClientState$) => ({ to, amount, asset, memo }: SendTxParams) =>
+const pushTx = (clientState$: BinanceClientState$) => ({ to, amount, asset, memo }: SendTxParams): Rx.Subscription =>
   tx$({ clientState$, to, amount, asset, memo }).subscribe(setTxRD)
 
+const sendStakeTx = (clientState$: BinanceClientState$) => ({ to, amount, asset, memo }: SendTxParams): TxLD =>
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/services/binance/transaction.ts:57]: Agree, we have to think about it again. We will do when we continue our all work for deposit txs in #537

File: src/renderer/services/bitcoin/transaction.ts [hunk 8]
```diff
@@ -39,6 +39,9 @@ const tx$ = ({ client$, to, amount, feeRate, memo }: { client$: Client$ } & Send
 const pushTx = (client$: Client$) => ({ to, amount, feeRate, memo }: SendTxParams) =>
   tx$({ client$, to, amount, feeRate, memo }).subscribe(setTxRD)
 
+const sendStakeTx = (client$: Client$) => ({ to, amount, feeRate, memo }: SendTxParams) =>
```
[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/services/bitcoin/transaction.ts:42]: Same questoin?

**Pull request merged**